apps/extension/client/src/checkTypegenNesting.ts

```diff
@@ -0,0 +1,150 @@
+import * as vscode from "vscode";
+import { WorkspaceConfiguration } from "vscode";
+
+const typeGenPatternKey = "*.ts";
```

Comment:
We also need to handle `.tsx` files. And TS 4.7 will also come with two new extensions - `.mts` and `.cts`. We don't necessarily support those 2 new extensions anywhere else right now so it's OK to skip them but perhaps we should add a TODO comment about adding support for them in the future or smth.

Reply:
Great input, I've added all four filetypes.